### PR TITLE
feat: Add global setTimeout API

### PR DESCRIPTION
**Pull request by berabhishek**
## Description

> This PR introduces the setTimeout feature for the button onClick call. 

Fixes https://github.com/appsmithorg/appsmith/issues/776

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?
Tets plan (WIP)
- [ ] https://github.com/appsmithorg/TestSmith/issues/1961
- [ ] https://github.com/appsmithorg/TestSmith/issues/2010

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by berabhishek** [comment]
Any update on this?

**Comment by sanjus-robotic-studio** [comment]
Hey, is there any update on this, as one of my clients have asked me to develop an app for them which requires set timeout feature. 

Thanks,
Sanjiv Anand

**Comment by sanjus-robotic-studio** [comment]
Hey, is there any update on this, as one of my clients have asked me to develop an app for them which requires set timeout feature. 

Thanks,
Sanjiv Anand

**Review by rimildeyjsr — DISMISSED** [review]
LGTM

**Comment by ramsaptami** [comment]
The DP here isn't working, returns a 404 page

**Comment by rimildeyjsr** [comment]
@berabhishek - can you pull in the latest release and rebase this branch once?

**Comment by berabhishek** [comment]
Added all changes from the release branch to feature/settimeout

File: app/client/src/sagas/ActionExecution/SetTimeoutSaga.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        dynamicString: `{{${callback}}}`,
+        // pass empty object to execute it as a callback function
+        callbackData: [{}],
+        event: { type: eventType },
+        triggerPropertyName: triggerMeta.triggerPropertyName,
+        source: triggerMeta.source,
+      });
+    } catch (e) {
+      logActionExecutionError(
+        e.message,
```
Comment (ohansFavour) [inline app/client/src/sagas/ActionExecution/SetTimeoutSaga.ts:56]: @berabhishek , as `e` is unknown, could you instead do this instead? 

(e as Error).message

Comment (berabhishek) [reply app/client/src/sagas/ActionExecution/SetTimeoutSaga.ts]: added

**Comment by berabhishek** [comment]
Label checker is failing after merging the changes from master into the current branch. Can anyone suggest what to do on these changes?

**Comment by ohansFavour** [comment]
> Label checker is failing after merging the changes from master into the current branch. Can anyone suggest what to do on these changes?

@berabhishek After QA verifies your PR, a label will be added after which the “Label checker” check will pass.

cc: @ramsaptami

**Comment by berabhishek** [comment]
Hey because this PR is moving very slowly, everytime someone makes a change in the master I am seeing conflicts. can someone pull this request resolve this change and move this faster? 

let me know if there are any better way to do this?

**Comment by ramsaptami** [comment]
Apologies for this @berabhishek, there's an issue with the DP deployment hence we're unable to test it the usual way we do here and that's causing a hiccup. We'll target to finish QA early this week so we get moving on this PR ASAP.

**Comment by ramsaptami** [comment]
@berabhishek there are a few fixes needed for the following scenarios:
- [ ] Need an id field for `clearTimeout` as you see for `clearInterval`
image
- [ ] clearInterval does not work - the function does not identify the id being passed and comes up as _undefined_
- [ ] We need error handing when timeout duration is set in string and timer is triggered

**Comment by taboxdev** [comment]
Folks , timeout is a core functionality, any updates on this ?

**Comment by ramsaptami** [comment]
@taboxdev: You can track setTimeout implementation [here](https://github.com/appsmithorg/appsmith/pull/17445). It should be out for your use in the next few releases.

**Comment by satbir121** [comment]
Closing the PR due to inactivity.

**Pull request closed without merging**